frontend/packages/console-metal3/src/components/host.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import {
+  BaremetalHostRole,
+  BaremetalHostStatus,
+  getHostMachineName,
+  getHostBmcAddress,
+  getName,
+  getNamespace,
+  getResource,
+  getSimpleHostStatus,
+  getUid
+} from 'kubevirt-web-ui-components';
+
+// TODO(jtomasek): update import once models are moved to console-shared package
+import { MachineModel, NodeModel } from '@console/internal/models';
+import { BaremetalHostModel } from '../models';
+
+import { referenceForModel } from '@console/internal/module/k8s';
+
+import {
+  ListHeader,
+  ColHead,
+  List,
+  ListPage,
+  ResourceRow
+} from '@console/internal/components/factory';
+
+import { ResourceLink } from '@console/internal/components/utils';
+import { WithResources } from '@console/shared';
+
+import MachineCell from './machine-cell';
+
+const nameColumnClasses = 'col-lg-2 col-md-4 col-sm-6 col-xs-6';
+const statusColumnClasses = 'col-lg-2 col-md-4 hidden-sm hidden-xs';
+const machineColumnClasses = 'col-lg-3 visible-lg';
+const roleColumnClasses = 'col-lg-2 visible-lg';
+const addressColumnClasses = 'col-lg-2 visible-lg';
+
+const HostHeader = props => (
+  <ListHeader>
+    <ColHead {...props} className={nameColumnClasses} sortField="metadata.name">
+      Name
+    </ColHead>
+    <ColHead {...props} className={statusColumnClasses}>
+      Status
+    </ColHead>
+    <ColHead {...props} className={machineColumnClasses}>
+      Machine
+    </ColHead>
+    <ColHead {...props} className={roleColumnClasses}>
+      Role
+    </ColHead>
+    <ColHead
+      {...props}
+      className={addressColumnClasses}
+      sortField="spec.bmc.address"
+    >
+      Management Address
+    </ColHead>
+  </ListHeader>
+);
+
+const HostRow = ({ obj: host }) => {
+  const name = getName(host);
+  const namespace = getNamespace(host);
+  const uid = getUid(host);
+  const machineName = getHostMachineName(host);
+  const address = getHostBmcAddress(host);
+
+  const machineResource = {
+    kind: referenceForModel(MachineModel),
+    name: machineName,
+    namespaced: true,
+    namespace,
+    isList: false,
+    prop: 'machine'
+  };
+
+  const hostResourceMap = {
+    machine: {
+      resource: machineResource
+    },
+    nodes: {
+      resource: getResource(NodeModel, { namespaced: false })
+    }
+  };
+
+  return (
+    <ResourceRow obj={host}>
+      <div className={nameColumnClasses}>
+        <ResourceLink
+          kind={BaremetalHostModel.kind}
+          name={name}
+          namespace={namespace}
+          title={uid}
+        />
+      </div>
+      <div className={statusColumnClasses}>
+        <WithResources resourceMap={machineName ? hostResourceMap : {}}>
```

Comment:
We should not be opening WebSockets for individual table cells

Reply:
I remember addressing this kind of issue in our Console fork, @rawagner can you please share the details?

Reply:
We dealt with this on multiple places, i.e. to show VM status which depends on multiple referenced objects. Optimized by requesting larger dataset via lower count of WebSockets and filtering on UI.